lib/measured/unit.rb

```diff
@@ -40,6 +59,11 @@ def inverse_conversion_amount
 
   private
 
+  def with_case_sensitivity(comparison)
+    comparison = comparison.is_a?(Array) ? comparison.map{|c| c.downcase} : comparison.downcase if !case_sensitive
```

Comment:
The other class it could be is an instance of a String, so I would also have to make the comparison an Array if it is not case_sensitive as well so I can use #include? . 

How about:
```ruby
def with_case_sensitivity(comparison)
  comparison = [comparison].flatten
  case_sensitive ? comparison : comparison.map(&:downcase)
end
```
Then I'd change #name_eql to use #include? in the comparison.